app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java

```diff
@@ -1033,6 +1048,76 @@ public void onCompareProductButtonClick() {
         startActivity(intent);
     }
 
+    @OnClick(R.id.btn_eating)
+    public void onBtnEatingClick() {
+        btnEating.setVisibility(View.GONE);
+        btnEatingTicked.setVisibility(View.VISIBLE);
+        addProductToEatenList();
+    }
+
+    @OnClick(R.id.btn_eating_ticked)
+    public void onBtnEatingTickedClick() {
+        btnEating.setVisibility(View.VISIBLE);
+        btnEatingTicked.setVisibility(View.GONE);
+        removeProductFromEatenlist();
+    }
+
+    public boolean checkIfProductIsEaten() {
+        List<YourListedProduct> eatenProducts =
+                yourListedProductsDao._queryProductLists_Products(1L);
+        if (!eatenProducts.isEmpty()) {
+            for (YourListedProduct product : eatenProducts) {
+                if (product.getBarcode().equals(barcode)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public void addProductToEatenList() {
+        String barcode = product.getCode();
+        String productName = product.getProductName();
+        StringBuilder stringBuilder = new StringBuilder();
+        if (isNotEmpty(product.getBrands())) {
+            stringBuilder.append(product.getBrands());
+        }
+        if (isNotEmpty(product.getQuantity())) {
+            stringBuilder.append(" - ").append(product.getQuantity());
+        }
+        String productDetailsString = stringBuilder.toString();
+        String imageUrl = product.getImageUrl();
+        if (isNotEmpty(product.getImageUrl())) {
```

Comment:
Lines 1090 -1092 should be removed as not usefull ( the stringBuilder is no more used, productDetailsString is already built and the image URL is set below)